presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -492,6 +493,20 @@ public ConnectorSession toConnectorSession(ConnectorId connectorId)
                 sessionPropertyManager);
     }
 
+    public ConnectorSession toConnectorSession(ConnectorId connectorId, WarningCollector warningCollector)
+    {
+        requireNonNull(connectorId, "connectorId is null");
+
+        return new FullConnectorSession(
+                this,
+                identity.toConnectorIdentity(connectorId.getCatalogName()),
+                connectorProperties.getOrDefault(connectorId, ImmutableMap.of()),
+                connectorId,
+                connectorId.getCatalogName(),
+                sessionPropertyManager,
+                warningCollector);
+    }
```

Comment:
warning collector will be "final" in session; so you don't need this.

Reply:
Deleted as we include warningCollector in session now.